src/domain/schemas/document-schemas.js

```diff
@@ -15,17 +15,31 @@ export const patchDocSectionsBodySchema = joi.object({
   sections: joi.array().items(sectionSchema)
 });
 
+export const publicContextSchema = joi.object({
+  archived: joi.boolean().required(),
+  verified: joi.boolean().required(),
+  review: joi.string().allow(null).allow('').required(),
+  allowedOpenContribution: joi.string().valid(...Object.values(DOCUMENT_ALLOWED_OPEN_CONTRIBUTION)).required()
+});
+
+export const publicContextConditionalSchema = joi.alternatives().conditional(
+  'roomId',
+  {
+    is: idOrKeySchema,
+    then: null,
+    otherwise: publicContextSchema.required()
+  }
```

Comment:
For some reason I would have expected the inverted test:

~~~
{
  is: null,
  then: publicContextSchema.required(),
  otherwise: null
}
~~~

But it's probably a matter of taste. Consider this comment purely informational and non-passionate :smile:

Reply:
This is just temporary, it will make sense when we add the schema for `roomContext`, in which case `is: idOrKeySchema` is the setup that makes more sense.

Reply:
Actually it's been long enough that I forgot the code. This is a dedicated schema for public context only.. In which case what you suggested is more logical. I am changing this already in the follow-up ticket.

Reply:
Poifect!